### PR TITLE
Partner Block and Release - TSV format

### PR DESCRIPTION
**Pull request by av-ps**

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	public PInstanceId getUnprocessedRowsSelectionId(@NonNull final BPartnerBlockFileId bPartnerBlockFileId)
+	{
+		return Optional.ofNullable(getUnprocessedActiveRowsQuery(bPartnerBlockFileId).createSelection())
+				.orElseThrow(() -> new AdempiereException("No unprocessed rows found for " + bPartnerBlockFileId));
+	}
+
+	public boolean isImported(@NonNull final BPartnerBlockFileId fileId)
+	{
+		return queryBL.createQueryBuilder(I_I_BPartner_BlockStatus.class)
+				.addEqualsFilter(I_I_BPartner_BlockStatus.COLUMNNAME_C_BPartner_Block_File_ID, fileId)
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java:71]: shall be in repository

File: backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	{
+		return getUnprocessedActiveRowsQuery(fileId)
+				.firstId() > 0;
+	}
+
+	@NonNull
+	private IQuery<I_I_BPartner_BlockStatus> getUnprocessedActiveRowsQuery(@NonNull final BPartnerBlockFileId fileId)
+	{
+		return queryBL.createQueryBuilder(I_I_BPartner_BlockStatus.class)
+				.addOnlyActiveRecordsFilter()
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java:87]: same, shall be in repository, not in service

File: backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/blockstatus/BPartnerBlockStatusImportProcess.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+				.forEach(bPartnerBlockStatusService::createBPartnerBlockStatus);
+
+		return ImportRecordResult.Inserted;
+	}
+
+	@Override
+	protected String getImportOrderBySql()
+	{
+		return I_I_BPartner_BlockStatus.COLUMNNAME_I_BPartner_BlockStatus_ID;
+	}
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/blockstatus/BPartnerBlockStatusImportProcess.java:107]: Why not per partner?

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessInfo.java [hunk 4]
```diff
@@ -408,6 +408,13 @@ public TableRecordReference getRecordRefOrNull()
 		return TableRecordReference.of(adTableId, recordId);
 	}
 
+	@NonNull
+	public TableRecordReference getRecordRefNotNull()
+	{
+		return Optional.ofNullable(getRecordRefOrNull())
+				.orElseThrow(() -> new AdempiereException("No TableRecordReference! PInstanceID=" + getPinstanceId()));
+	}
```
Comment (cristinamghita) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessInfo.java:416]: Pls add a translated message

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/attachments/AttachmentEntryService.java [hunk 5]
```diff
@@ -396,6 +397,24 @@ public void updateData(
...
 
+	@NonNull
+	public Optional<AttachmentEntry> getUniqueByReferenceRecord(@NonNull final TableRecordReference recordReference)
+	{
+		final List<AttachmentEntry> attachmentEntries = getByReferencedRecord(recordReference);
+
+		if (attachmentEntries.size() > 1)
+		{
+			throw new AdempiereException("Expected to find only one attachment entry for the given TableRecordReference, but multiple entries matched!")
+					.appendParametersToMessage()
```
Comment (cristinamghita) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/attachments/AttachmentEntryService.java:408]: Pls, add a translated message

**Review by cristinamghita — CHANGES_REQUESTED** [review]
Pls check my comments

File: backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/blockstatus/BPartnerBlockStatusImportProcess.java (hunk 3 above)
Comment (cp-ps) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/blockstatus/BPartnerBlockStatusImportProcess.java:107]: as `I_BPartner_BlockStatus_ID` is the best indicator we have for what the order of the rows was in the imported file... that way we have a predictable outcome.

File: backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java (hunk 2 above)
Comment (cp-ps) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/blockstatus/file/BPartnerBlockFileService.java:87]: introduce `IBPartnerBlockStatusDAO` - IMHO we should keep Repositories for domain objects like `BPartnerBlockFile`

**Review by cristinamghita — APPROVED** [review]
lgtm!

**Pull request merged**